swri_rospy/src/swri_rospy/__init__.py

```diff
@@ -62,3 +63,32 @@ def wrapper(*args, **kwds):
             rospy.logerr(traceback.format_exc())
             return None
     return wrapper
+
+
+def subscribe_single_threaded(name, data_class, callback=None,
```

Comment:
I prefer subclassing over class-wrapping, especially in Python, especially after I watched [Super Considered Super (youtube link)](https://youtu.be/EiOglTERPEo), which is a fun talk that you'd probably like.